src/SharedLibrary.cpp

```diff
@@ -55,55 +55,63 @@ bool SharedLibrary::open(const char *filename)
     close();
 #if defined(_WIN32)
     mPriv->implementation = (void*)LoadLibrary(filename);
-    LPTSTR msg = nullptr;
-    FormatMessage(
-       FORMAT_MESSAGE_FROM_SYSTEM |FORMAT_MESSAGE_ALLOCATE_BUFFER | FORMAT_MESSAGE_IGNORE_INSERTS,
-       nullptr, GetLastError(), MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT),
-       (LPTSTR)&msg, 0, nullptr);
+    if (!mPriv->implementation) {
+        LPTSTR msg = nullptr;
+        FormatMessage(
+            FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_ALLOCATE_BUFFER | FORMAT_MESSAGE_IGNORE_INSERTS,
+            nullptr, GetLastError(), MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT),
```

Comment:
GetLastError() is returning "The operation completed successfully", so checking if msg != nullptr is not enough to check for errors.